### PR TITLE
added filter on databaseView that filters databases based on access in roles

### PR DESCRIPTION
**Pull request by pbr0ck3r**

### CODE REVIEW

**Comment by xrmx** [comment]
Looks like there's a lot of unwanted changes in this PR

**Comment by pbr0ck3r** [comment]
If you could explained to me the unwanted to changes or the way you would prefer I implement the filter that would be helpful. I previously just followed @mistercrunch directions that were suggested (https://github.com/apache/incubator-superset/issues/5207#issuecomment-436886833).

**Comment by pbr0ck3r** [comment]
@xrmx I believe I understand what you mean by unwanted changes. Will remove unwanted file changes.

File: superset/views/core.py [hunk 1]
```diff
@@ -93,6 +93,15 @@ def apply(self, query, func):  # noqa
         return query.filter(self.model.perm.in_(perms))
 
 
+class DatabaseFilter(SupersetFilter):
+    def apply(self, query, func):  # noqa
+        if security_manager.all_database_access():
+            return query
+        perms = self.get_view_menus('database_access')
+        # TODO(bogdan): add `schema_access` support here
```
Comment (mistercrunch) [inline superset/views/core.py]: Let's remove that TODO line.

**Comment by mistercrunch** [comment]
Minor comment, otherwise LGTM

File: superset/views/core.py (hunk 1 above)
Comment (pbr0ck3r) [reply superset/views/core.py]: Fixed!

**Review by mistercrunch — CHANGES_REQUESTED** [review]
Oh one more thing, we should make sure that role `Alpha` sees all databases as well for backwards compatibility purposes.
You'll need to add something here:
https://github.com/apache/incubator-superset/blob/master/superset/security.py#L87

And remove this line:
https://github.com/apache/incubator-superset/blob/master/superset/security.py#L71

Would be great to have a unit test validating that:
1. Alpha sees all dbs when calling `/databaseview/api/read`
2. Gamma sees no dbs when calling `/databaseview/api/read`

**Comment by pbr0ck3r** [comment]
@mistercrunch I went ahead and made the suggestion changes and also updated the tests accordingly.

File: superset/security.py [hunk 2]
```diff
@@ -111,6 +111,10 @@ def all_datasource_access(self):
     def all_database_access(self):
         return self.can_access('all_database_access', 'all_database_access')
 
+    def all_database_access(self):
```
Comment (xrmx) [inline superset/security.py]: Isn't this the very same method as the one above?

Comment (pbr0ck3r) [reply superset/security.py]: Yes it is. Went ahead and removed it. Must have been duplicated when I was merging in new changes from master into my branch.

**Comment by tomaszj** [comment]
Hi folks, is there anything that can be contributed to integrate the changes from this PR?

**Review by mistercrunch — APPROVED** [review]

**Comment by mistercrunch** [comment]
Please rebase / fix conflicts

**Comment by pbr0ck3r** [comment]
Not stale

**Comment by mistercrunch** [comment]
A similar PR has been merged already, closing https://github.com/apache/incubator-superset/blob/8c17c1927937362a22c07d1362f00992e56621a1/superset/views/core.py#L203-L208

**Comment by pbr0ck3r** [comment]
Kind of salty that this PR has been open for 8 months and changes that were asked were fixed months ago as well. But I guess if someone else beat me to the punch of getting the same changes in... glad the security issue is fixed.

**Comment by mistercrunch** [comment]
@pbr0ck3r it's hard for maintainers to keep up with the pace of this repository and the volume of information that gets through here. I wish all PRs and issues would get the attention they deserve. 

But I don't think I get notifications when new commits are added to a PR, so I'd recommend commenting/tagging on the PR once comments are addressed. I process most of my notifications, though I'll miss a certain percentage, but if there's no notifications there's no chance maintainers will catch things.

**Pull request closed without merging**